samples/azure-quantum/parallel-qrng/ParallelQrng.ipynb

```diff
@@ -459,7 +459,7 @@
``\\r\\nIn []: %trace MyOperation\\r\\nOut[]: <HTML visualization of the operation>\\r\\n```\\r\\n                    \",\"\\r\\nVisualize the execution path of a Q# operation defined as `operation MyOperation(a : Int, b : Int) : Result`:\\r\\n```\\r\\nIn []: %trace MyOperation a=5 b=10\\r\\nOut[]: <HTML visualization of the operation>\\r\\n```\\r\\n                    \",\"\\r\\nVisualize operations at depth 2 on the execution path of a Q# operation defined\\r\\nas `operation MyOperation() : Result`:\\r\\n```\\r\\nIn []: %trace MyOperation --depth=2\\r\\nOut[]: <HTML visualization of the operation>\\r\\n```\\r\\n                    \"],\"SeeAlso\":null},\"AssemblyName\":\"Microsoft.Quantum.IQSharp.Kernel\"},{\"Name\":\"%who\",\"Documentation\":{\"Summary\":\"Lists the Q# operations available in the current session.\",\"Full\":null,\"Description\":\"\\r\\nThis magic command returns a list of Q# operations and functions that are available\\r\\nin the current IQ# session for use with magic commands such as `%simulate`\\r\\nand `%estimate`.\\r\\n\\r\\nThe list will include Q# operations and functions which have been defined interactively\\r\\nwithin cells in the current notebook (after the cells have been executed),\\r\\nas well as any Q# operations and functions defined within .qs files in the current folder.\\r\\n                \",\"Remarks\":null,\"Examples\":[\"\\r\\nDisplay the list of Q# operations and functions available in the current session:\\r\\n```\\r\\nIn []: %who\\r\\nOut[]: <list of Q# operation and function names>\\r\\n```\\r\\n                    \"],\"SeeAlso\":null},\"AssemblyName\":\"Microsoft.Quantum.IQSharp.Kernel\"},{\"Name\":\"%workspace\",\"Documentation\":{\"Summary\":\"Provides actions related to the current workspace.\",\"Full\":null,\"Description\":\"\\r\\nThis magic command allows for displaying and reloading the Q# operations and functions\\r\\ndefined within .qs files in the current folder.\\r\\n\\r\\nIf no parameters are provided, the command displays a list of Q# operations or functions\\r\\nwithin .qs files in the current folder which are available\\r\\nin the current IQ# session for use with magic commands such as `%simulate`\\r\\nand `%estimate`.\\r\\n\\r\\nThe command will also output any errors encountered while compiling the .qs files\\r\\nin the current folder.\\r\\n\\r\\n#### Optional parameters\\r\\n\\r\\n- `reload`: Causes the IQ# kernel to recompile all .qs files in the current folder.\\r\\n                \",\"Remarks\":null,\"Examples\":[\"\\r\\nDisplay the list of Q# operations and functions available in the current folder:\\r\\n```\\r\\nIn []: %workspace\\r\\nOut[]: <list of Q# operation and function names>\\r\\n```\\r\\n                    \",\"\\r\\nRecompile the .qs files in the current folder:\\r\\n```\\r\\nIn []: %workspace reload\\r\\nOut[]: <list of Q# operation and function names>\\r\\n```\\r\\n                    \"],\"SeeAlso\":null},\"AssemblyName\":\"Microsoft.Quantum.IQSharp.Kernel\"}]",
```

Comment:
It looks from the diffs as though the version number is replaced in the outputs, but without rerunning the notebook?

Reply:
Chris, I have no experience with this repo. Can you please clarify a bit more what I need to do? (continue/correct the following list)
* Clone the repo
* Checkout `kuzminrobin/hotfix.0.18.2108.160999` branch
* cd samples/azure-quantum/parallel-qrng
* Open `ParallelQrng.ipynb`
* Run all the cells.
* \<Close `ParallelQrng.ipynb`\>
* Commit the changes in `ParallelQrng.ipynb`.

Reply:
I'm trying to run `ParallelQrng.ipynb`.

I follow these instructions: [Q# with Jupyter Notebook](https://github.com/microsoft/Quantum/tree/main/samples/azure-quantum/parallel-qrng#q-with-jupyter-notebook).

I installed Anaconda 2021.05.
In Anaconda Prompt did `conda init`.
In another Anaconda Prompt `conda init powershell cmd.exe`.
In another Anaconda Prompt 
* `conda create -n qsharp-env02 -c quantum-engineering qsharp notebook`.
* `conda activate qsharp-env02`
* `python -c "import qsharp"`
And I get:
```
(qsharp-env02) C:\Users\rokuzmin>python -c "import qsharp"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\rokuzmin\Anaconda3\envs\qsharp-env02\lib\site-packages\qsharp\__init__.py", line 198, in <module>
    client = _start_client()
  File "C:\Users\rokuzmin\Anaconda3\envs\qsharp-env02\lib\site-packages\qsharp\clients\__init__.py", line 40, in _start_client
    client.start()
  File "C:\Users\rokuzmin\Anaconda3\envs\qsharp-env02\lib\site-packages\qsharp\clients\iqsharp.py", line 92, in start
    self.kernel_manager.start_kernel(extra_arguments=["--user-agent", f"qsharp.py{_user_agent_extra}"])
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_client\manager.py", line 309, in start_kernel
    kernel_cmd, kw = self.pre_start_kernel(**kw)
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_client\manager.py", line 258, in pre_start_kernel
    self.write_connection_file()
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_client\connect.py", line 468, in write_connection_file
    self.connection_file, cfg = write_connection_file(self.connection_file,
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_client\connect.py", line 138, in write_connection_file
    with secure_write(fname) as f:
  File "C:\Users\rokuzmin\Anaconda3\envs\qsharp-env02\lib\contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_core\paths.py", line 461, in secure_write
    win32_restrict_file_to_user(fname)
  File "C:\Users\rokuzmin\AppData\Roaming\Python\Python39\site-packages\jupyter_core\paths.py", line 387, in win32_restrict_file_to_user
    import win32api
ImportError: DLL load failed while importing win32api: The specified procedure could not be found.
```
What do I miss?

Reply:
> What do I miss?

I'm not sure what may be causing that error, in particular it seems internal to the jupyter-core Python package that's part of the Jupyter Notebook application.

> Can you please clarify a bit more what I need to do? (continue/correct the following list)
>
> * Clone the repo
> * Checkout `kuzminrobin/hotfix.0.18.2108.160999` branch
> * cd samples/azure-quantum/parallel-qrng
> * Open `ParallelQrng.ipynb`
> * Run all the cells.
> * \<Close `ParallelQrng.ipynb`\>
> * Commit the changes in `ParallelQrng.ipynb`.

This looks correct, aside from that we'll need to do so for each ipynb (some have separate environments, since they have requirements beyond the base requirements for the QDK). I'm a bit confused as to why those steps resulting in the outputs having the exact same timing, let me look a little more.

Reply:
Ah, I see, I think I may have misunderstood something in the diff, sorry. Dismissing my review.

Reply:
OK to merge? If yes then can you please approve?